lib/core/src/chain_swap.rs

```diff
@@ -436,6 +436,32 @@ impl ChainSwapHandler {
             | ChainSwapStates::TransactionLockupFailed
             | ChainSwapStates::TransactionRefunded
             | ChainSwapStates::SwapExpired => {
+                // Zero-amount Receive Chain Swaps also get to TransactionLockupFailed when user locks up funds
+                let is_zero_amount = swap.payer_amount_sat == 0;
+                if matches!(swap_state, ChainSwapStates::TransactionLockupFailed) && is_zero_amount
+                {
+                    let accept_quote_res = self
+                        .swapper
+                        .get_zero_amount_chain_swap_quote(&swap.id)
+                        .map(|quote| quote.to_sat())
+                        .and_then(|quote| {
+                            info!("Got quote of {quote} sat for swap {}", &swap.id);
+
+                            self.persister
+                                .update_zero_amount_swap_values(&swap.id, quote)?;
```

Comment:
What does the quote represent here? Is it the boltz lockup? If so how do we know they don't cheat us with extra service fee? Don't we need to validate that?

Reply:
The quote represents the server lockup amount, i.e. "this is how much you can claim, if you choose to accept this quote".

How we normally validate the fees for Receive Chain Swaps is during creation:
https://github.com/breez/breez-sdk-liquid/blob/98864c70a6055c74d5b60142548de8a9584b7b93/lib/core/src/sdk.rs#L1913-L1916

However, for this the swap amount has to be known on creation, and in this case it's not.

I didn't add validation because we cannot do it with the data we know.

To be able to do this for zero-amount swaps, we would need:
- to store the swap's fee pair details (claim fee estimate, server fee, service fee rate)
- to fetch the actual user lockup amount from the swap's lockup address
- to calculate the expected server lockup amount and compare with the quote

Should I add this?

Reply:
I think we don't have a choice, otherwise we are committing to a swap by trusting the swapper only.
By doing so we will also be able to calculate properly the payer amount.

Reply:
Done, added validation before accepting the quote from Boltz.

We also now correctly persist the payer and receiver amounts.